### ISSUE TITLE
MultiMC Hangs on Update

### ISSUE THREAD

**Issue by MggMuggins** [MultiMC/MultiMC5#1702]
System Information
-----------------------------
MultiMC version: 0.4.10-684

Operating System: Ubuntu 16.04 LTS

Summary of the issue or suggestion:
----------------------------------------------
I've been having trouble trying to update MultiMC for ages, but I've finally figured that it's time to fix it. I had a little trouble with slow internet and my browser clogging things up, but I think I've fixed that (no more errors during update, so I assume it's gone). I'm currently running MultiMC without a functional .desktop file (which is fine, for now), using the command line, so I can see all of the program output. After I've started MultiMC, I ccan check for updates just fine. That works quite quickly (which of course, other than downloads, is the standard I've come to expect from this wonderful little piece of software). However, I've pressed the "Update Now" button, MultiMC opens an update window, which seems to work for a while. The terminal plops out output at a fairly steady rate, and everything seems to run just fine. However, after most of the stuff has apparently done downloading, the GUI freezes and has to be killed, and the terminal continues with its status checks. I'll post a section of the program log below.

What should happen:
------------------------------
At this point, given all the "Finished" that I see in the terminal, the download bar should close, and I would expect another pop-up that prompts me to restart MultiMC so that it can finish updating. Of course, this doesn't happen.

Steps to reproduce the issue (Add more if needed):
-------------------------------------------------------------
I'm not sure if there are any steps that anyone would be able to take, but I'll basically list my setup:

1. Use a very old MultiMC folder (probably close to two years, updated a few times), that is quite large with a lot of things. 
2. Run on Linux, without using the .desktop file. Always run from the command line. The .desktop is broken anyway, so this is how I have to do it. (I used to have one set up, but then I had to do a fresh OS install...)

Suspected cause:
---------------------------
Probably the fact that I'm running from the terminal, but really I have no idea. I suspect that the window just isn't being shut down properly, which causes me to have to kill the program. If that window would close right, I think that updating would work.

Logs/Screenshots:
----------------------------
```   19.977 D Downloading  "http://files.multimc.org/files/1bbde436a658e17a50ed38086c06a5f2-JavaCheck.jar"  local MD5:  ""
   19.977 D Expecting  "1bbde436a658e17a50ed38086c06a5f2"
   19.977 D Downloading  "http://files.multimc.org/files/23c0368af761653bb8bb15c3e5c615df-MultiMC"  local MD5:  ""
   19.977 D Expecting  "23c0368af761653bb8bb15c3e5c615df"
   19.977 D Downloading  "http://files.multimc.org/files/103de5a0ebeb4cf9ceaf012b35839fb1-NewLaunch.jar"  local MD5:  ""
   19.977 D Expecting  "103de5a0ebeb4cf9ceaf012b35839fb1"
   19.977 D Downloading  "http://files.multimc.org/files/9a348c72c206e48d0551451eb66d257e-libMultiMC_logic.so"  local MD5:  ""
   19.977 D Expecting  "9a348c72c206e48d0551451eb66d257e"
   20.389 D Finished  "http://files.multimc.org/files/1bbde436a658e17a50ed38086c06a5f2-JavaCheck.jar"  got  "1bbde436a658e17a50ed38086c06a5f2"
   21.346 D Finished  "http://files.multimc.org/files/103de5a0ebeb4cf9ceaf012b35839fb1-NewLaunch.jar"  got  "103de5a0ebeb4cf9ceaf012b35839fb1"
   60.323 D "Status JSON"  started.
   60.323 D Downloading  "http://status.mojang.com/check"
   61.185 D "Status JSON" succeeded.
   61.185 D Task succeeded
   61.185 D Finished loading status JSON.
   61.186 D Status loading succeeded.
   97.942 D Finished  "http://files.multimc.org/files/23c0368af761653bb8bb15c3e5c615df-MultiMC"  got  "23c0368af761653bb8bb15c3e5c615df"
  120.328 D "Status JSON"  started.
  120.328 D Downloading  "http://status.mojang.com/check"
  120.404 D "Status JSON" succeeded.
  120.404 D Task succeeded
  120.404 D Finished loading status JSON.
  120.404 D Status loading succeeded.
  180.324 D "Status JSON"  started.
  180.324 D Downloading  "http://status.mojang.com/check"
  180.401 D "Status JSON" succeeded.
  180.401 D Task succeeded
  180.401 D Finished loading status JSON.
  180.401 D Status loading succeeded.
```

Additional Info:
---------------------------
Three MD5 sums did not match while updating. I hope that doesn't have anything to do with it.

**Comment by peterix** [comment]
It looks like `http://files.multimc.org/files/9a348c72c206e48d0551451eb66d257e-libMultiMC_logic.so` didn't finish downloading, so it's stuck waiting for that.

**Comment by MggMuggins** [comment]
Thanks! How do you propose I solve the issue, without the ability to update...

**Comment by peterix** [comment]
You can grab the stable release on the website and move over your data to the new copy.

You should keep these:
![image](https://cloud.githubusercontent.com/assets/203326/19920770/0c71a0fa-a0db-11e6-94e7-3b887b0944b7.png)

That's probably a lot easier than figuring out why the updater doesn't work for you. I've fixed a lot of problems with downloads since 0.4.10, especially in the `develop` version.

**Comment by MggMuggins** [comment]
Sorry, but github's images aren't working for me (don't display, I think it's a Linux issue...). Any way you could write out the contents of that image?

**Comment by QuImUfu** [comment]
image url: 
https://cloud.githubusercontent.com/assets/203326/19920770/0c71a0fa-a0db-11e6-94e7-3b887b0944b7.png

Written:
in the MMC-Folder keep:
accounts
assets
cache 
icons
instances
libaries
lwjgl
mods
translations
versions
accounts.json
metacache
multimc.cfg
notifications.json

**Comment by MggMuggins** [comment]
The image URL doesn't work either, sadly. Thanks for this though. The rest of the folders in the newer version of MMC is going to be in other places than those folders, I'd assume.

**Comment by MggMuggins** [comment]
Now that I go to actually do the thing, the image works...
Anywho, thanks for the advice. I've installed the Ubuntu package (which didn't exist when I first got MMC), and I'm currently backing up my old MMC folder. Hopefully that all works out.

Another thing I'd like to say quick (I might open another Issue, who knows). The MMC .desktop files for ubuntu are a mess (at least for me, it is probably partially my fault), and they don't work. MMC only generates one that I can see in /usr/share/applications, but I see another in my unity menu (that doesn't launch MMC, just indicates that it's running). Running the one that MMC generated works just fine, but I can't get it to stay on my toolbar because it doesn't stick around long enough to click it. It goes away right after being run, and is replaced by the one that doesn't work.

Another issue that I've just now noticed (after installing the new version clean, and then importing my old stuff, is that the buttons and UI in general have gone pretty much crazy. Every button requires two clicks, and it's just a mess. I think actual MC is running fine though.

**Comment by peterix** [comment]
Well, I have Ubuntu 16.04 in a VM and it's not having any of those issues... So I wonder what you're doing different.

**Comment by peterix** [comment]
I'm going to fully update it and reinstall MultiMC on it just to make sure.

**Comment by peterix** [comment]
So, maybe check for any extra stray .desktop files? You could have one for the old MultiMC install that simply doesn't work and somehow gets assigned to the running process instead of the real one.

**Comment by peterix** [comment]
As for the UI being crazy... can't reproduce that problem.

**Comment by MggMuggins** [comment]
Mightn't it have something to do with some leftover config from some older version of MMC? I'll see how it behaves in the morning.

**Comment by QuImUfu** [comment]
I generally advice you not to use the .deb download but rather the default .tar.gz and create the .desktop file yourself. 
This way you can have the MMC-dir in a sane location like /home/${USER}/MultiMC. 

You Installation seems to behave completely crazy. 
Can you just create a new MultiMC install somewhere else (do not copy any files)  and try if this works?
If this does work you can copy just the instances folder. You would have to set up MMC again (account, password, Java settings), but should keep all instances.

If this doesn't work you could just boot from a live-cd (if you have enough (ca 4GB) ram with the toram kernel parameter. Warning: booting up with this param takes very long) and install MultiMC and try running it and see if everything in MultiMC except starting Minecraft (it is pretty likely the graphic won't work) works.

**Comment by peterix** [comment]
The .deb is fine and tested on 16.04... and I think `~/.local/share/multimc` is fine as well. If it bugs you that much, `ln -s ~/.local/share/multimc ~/MultiMC` solves that problem.

I suspect there is some second `.desktop` that doesn't come from the `.deb` and messes up the proper recognition of the process by unity.

**Comment by QuImUfu** [comment]
Jeah, it does work just fine, but i see no benefits in using the .deb, except for easy installation.  
But that is only my opinion (as some sort of linux nerd that likes to do a lot of things myself/in my way).   
Maybe i should care more what advice i give to people on the Internet, because i actually think the .deb is for many people the best way to go. Especially if you want things to just work. :)

**Comment by MggMuggins** [comment]
Well, thinking about it now, the buttons were an issue before I brought over my files, but now, having fired it up after restarting seems to have fixed that. The .desktops are still a problem, and my /usr/share/applications doesn't have anything that I could conceivably delete (here is the entire M/N section of that file):
![mn](https://cloud.githubusercontent.com/assets/16041213/19976563/35a56e00-a1be-11e6-90cc-7657812d5f18.png)
The two .desktops do have different icons (the one shown there is the one that works), but I have no idea where that second one would be. I did check, and there are no hidden files that would have anything to do with this issue.

Along the lines of the deb file, If I knew more about how all this Linux stuff works, and understood what I was doing with the tarball, I'd probably use that. However, the .deb is really easy, so I will usually go for that.

**Comment by QuImUfu** [comment]
is there anything about MultiMC in ${HOME}/.local/share/applications ?

**Comment by MggMuggins** [comment]
Found it. I'm going to try deleting that...
UPDATE: That did it! Thanks for the support guys! Just one more reason to absolutely adore this software! Keep doing what you're doing.

**Comment by peterix** [comment]
Alright! :)

The linux side of things will need some changes though...
See: https://github.com/MultiMC/MultiMC5/issues/1699

**Comment by MggMuggins** [comment]
I'm leaving MultiMC to run it's updates, but I'm continuing to have the same original problem, even after going through all the steps we did. I tell it to update, and it says "Downloading 7 update files", downloads 4 of them, and stalls. No more network activity from MultiMC. Log file: [http://pastebin.com/17Q6L90G](url)

**Comment by peterix** [comment]
Hm.

* I can't fix that if it can't update to the fixed version.
* It doesn't log enough to detemine what's actually going on.

So, the same advice as above applies.

If you want to help, it would be great if you could capture the network traffic during the update with Wireshark. Just make sure you shut down any other network accessing things while doing that.

**Comment by peterix** [comment]
Just to check if it's still a problem in the current version...

Can you grab the current release and switch to develop and back? Change the update channel to develop, update, change it to stable, update. Both **should** work.

**Comment by MggMuggins** [comment]
Sorry for the late response, I was diagnosing and fixing an issue with an atom package that I was trying to help out with.
Anyway, I grabbed a latest stable release off the website, and switched the update channel to development. When I checked for updates, it said there weren't any, but gave me the option to udate anyway. The dialogue popped up and started downloading one file. I'm experiencing the same symptoms with this as I am my copy.

A side note: Is there any particular reason that you can't do things while MultiMC is downloading stuff? I find this quite annoying, since my Internet is slower than anything you can comprehend, and I want to be able to work with my instances as I'm downloading stuff.

**Comment by peterix** [comment]
Are you sure it's not just horribly slow then?

You can think of everything in MultiMC as placeholders for something better. And I'm slowly replacing the placeholders. For example, the current version lets you launch multiple instances with all the windows still open. I just haven't done the same for downloads yet - and having no background tasks to keep track of made it easier to write.

Once I allow background downloads, I will have to have some intelligent download queuing and unification mechanism. 

* Multiple download tasks could require the same file and that has to be unified.
* Files that are required by multiple download tasks should have higher priority.
* There should be some UI for managing tasks and giving them priority or pausing them.
* ...and so on.

The current implementation seems to work just fine for most people and doesn't require any of this.

**Comment by peterix** [comment]
Anyway, I will add a bunch of logging calls to the develop version and ask you to grab it and downgrade it to stable. The logs should tell us more then.

**Comment by MggMuggins** [comment]
Alright. I'm pretty sure it's not slowness, since most of the bar goes at a reasonable rate, and then it slows down and pretty much halts. Also, I am running the gnome-system-monitor at the same time, so I can see network go from reasonable to nothing, at the same time the update halts.

**Comment by MggMuggins** [comment]
It finished, I think...
```0.761 D Following redirect to  "https://multimc.org/rss.xml"
    0.761 D Downloading  "https://multimc.org/rss.xml"
    0.761 D Download redirected: "https://multimc.org/rss.xml"
    0.773 D Download succeeded: "http://files.multimc.org/update/lin64/channels.json"
    0.773 D "Update System Channel List" succeeded.
    0.773 D Task succeeded
    0.773 D Successfully loaded UpdateChecker channel list.
    0.773 D Checking for updates.
    0.773 D m_repoUrl =  "http://files.multimc.org/update/lin64/develop/"
    0.773 D "GoUpdate Repository Index"  started.
    0.773 D Downloading  "http://files.multimc.org/update/lin64/develop/index.json"
    0.795 D Download succeeded: "http://status.mojang.com/check"
    0.796 D "Status JSON" succeeded.
    0.796 D Task succeeded
    0.796 D Finished loading status JSON.
    0.796 D Status loading succeeded.
    0.975 D Download succeeded: "http://files.multimc.org/notifications.json"
    0.978 D "Checking for notifications" succeeded.
    0.978 D Task succeeded
    1.723 D Download succeeded: "http://files.multimc.org/update/lin64/develop/index.json"
    1.723 D "GoUpdate Repository Index" succeeded.
    1.723 D Task succeeded
    1.723 D Finished downloading repo index. Checking for new versions.
    1.723 D Processing repository version list.
    1.754 D Loading remote version list.
    1.755 D Loaded version "17w06a" from remote version list.
    1.755 D Loaded version "1.11.2" from remote version list.
    1.755 D Loaded version "1.11.1" from remote version list.
    1.755 D Loaded version "16w50a" from remote version list.
    1.755 D Loaded version "1.11" from remote version list.
    1.755 D Loaded version "16w44a" from remote version list.
    1.755 D Loaded version "16w43a" from remote version list.
    1.755 D Loaded version "16w42a" from remote version list.
    1.755 D Loaded version "16w41a" from remote version list.
    1.755 D Loaded version "16w40a" from remote version list.
    1.755 D Loaded version "16w39c" from remote version list.
    1.755 D Loaded version "16w38a" from remote version list.
    1.755 D Loaded version "16w36a" from remote version list.
    1.755 D Loaded version "16w35a" from remote version list.
    1.755 D Loaded version "16w33a" from remote version list.
    1.755 D Loaded version "16w32b" from remote version list.
    1.755 D Loaded version "1.10.2" from remote version list.
    1.755 D Loaded version "1.10.1" from remote version list.
    1.755 D Loaded version "1.10" from remote version list.
    1.756 D Loaded version "1.9.4" from remote version list.
    1.756 D Loaded version "1.9.3" from remote version list.
    1.756 D Loaded version "1.9.2" from remote version list.
    1.756 D Loaded version "1.9.1" from remote version list.
    1.756 D Loaded version "1.9" from remote version list.
    1.756 D Loaded version "1.8.9" from remote version list.
    1.756 D Loaded version "1.8.8" from remote version list.
    1.756 D Loaded version "1.8.7" from remote version list.
    1.756 D Loaded version "1.8.6" from remote version list.
    1.756 D Loaded version "1.8.5" from remote version list.
    1.756 D Loaded version "1.8.4" from remote version list.
    1.756 D Loaded version "1.8.3" from remote version list.
    1.756 D Loaded version "1.8.2" from remote version list.
    1.756 D Loaded version "1.8.1" from remote version list.
    1.756 D Loaded version "1.8" from remote version list.
    1.756 D Loaded version "1.7.10" from remote version list.
    1.756 D Loaded version "1.7.9" from remote version list.
    1.756 D Loaded version "1.7.8" from remote version list.
    1.756 D Loaded version "1.7.7" from remote version list.
    1.756 D Loaded version "1.7.6" from remote version list.
    1.756 D Loaded version "1.7.5" from remote version list.
    1.756 D Loaded version "1.7.4" from remote version list.
    1.756 D Loaded version "1.7.3" from remote version list.
    1.756 D Loaded version "1.7.2" from remote version list.
    1.756 D Loaded version "1.6.4" from remote version list.
    1.756 D Loaded version "1.6.2" from remote version list.
    1.757 D Loaded version "1.6.1" from remote version list.
    1.758 W Blacklisted legacy version ignored:  "rd-160052"
    1.759 D Task succeeded
    1.812 D Download succeeded: "https://multimc.org/rss.xml"
    1.812 D "News RSS Feed" succeeded.
    1.812 D Task succeeded
    1.812 D Finished loading RSS feed.
    1.813 D Loaded news entry "MultiMC 0.5.1 - Bugfixes"
    1.813 D Loaded news entry "MultiMC 0.5.0 - Running multiple instances at the same time!"
    1.813 D Loaded news entry "Update to 0.4.11 - OSX 10.7 fix and workaround for Minecraft 1.9 compatibility"
    1.814 D Loaded news entry "Update to 0.4.10 - Another wifi fix"
    1.814 D Loaded news entry "Update to 0.4.9 - the Windows wifi fix"
    1.814 D Loaded news entry "Update to 0.4.8"
    1.814 D Loaded news entry "Update to 0.4.7, what 0.4.6 should have been."
    1.814 D Loaded news entry "Update to 0.4.6, Bug fixes and improvements!"
    1.814 D Loaded news entry "Update to 0.4.5, Translations, Themes, and Technic"
    1.814 D Loaded news entry "Version 0.4 out - Jar mods, UI changes, and more!"
    1.814 D News loading succeeded.
    3.229 D Download succeeded: "https://sourceforge.net/projects/java-game-lib/rss"
    3.229 D "LWJGL RSS" succeeded.
    3.229 D Task succeeded
    3.233 D Discovered LWGL version "2.9.3" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.9.3/lwjgl-2.9.3.zip/download"
    3.233 D Discovered LWGL version "2.9.2" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.9.2/lwjgl-2.9.2.zip/download"
    3.233 D Discovered LWGL version "2.9.1" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.9.1/lwjgl-2.9.1.zip/download"
    3.233 D Discovered LWGL version "2.9.0" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.9.0/lwjgl-2.9.0.zip/download"
    3.233 D Discovered LWGL version "2.8.5" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.8.5/lwjgl-2.8.5.zip/download"
    3.233 D Discovered LWGL version "2.8.4" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.8.4/lwjgl-2.8.4.zip/download"
    3.233 D Discovered LWGL version "2.8.3" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.8.3/lwjgl-2.8.3.zip/download"
    3.233 D Discovered LWGL version "2.8.2" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.8.2/lwjgl-2.8.2.zip/download"
    3.234 D Discovered LWGL version "2.8.1" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.8.1/lwjgl-2.8.1.zip/download"
    3.234 D Discovered LWGL version "2.8.0" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.8.0/lwjgl-2.8.0.zip/download"
    3.234 D Discovered LWGL version "2.7.1" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.7.1/lwjgl-2.7.1.zip/download"
    3.234 D Discovered LWGL version "2.7" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.7/lwjgl-2.7.zip/download"
    3.234 D Discovered LWGL version "2.6" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.6/lwjgl-2.6.zip/download"
    3.234 D Discovered LWGL version "2.5" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.5/lwjgl-2.5.zip/download"
    3.234 D Discovered LWGL version "2.4.2" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.4.2/lwjgl-2.4.2.zip/download"
    3.234 D Discovered LWGL version "2.4.1" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.4.1/lwjgl-2.4.1.zip/download"
    3.234 D Discovered LWGL version "2.4" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.4/lwjgl-2.4.zip/download"
    3.234 D Discovered LWGL version "2.3" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.3/lwjgl-2.3.zip/download"
    3.234 D Discovered LWGL version "2.2.2" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.2.2/lwjgl-2.2.2.zip/download"
    3.234 D Discovered LWGL version "2.2.1" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.2.1/lwjgl-2.2.1.zip/download"
    3.234 D Discovered LWGL version "2.2.0" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.2.0/lwjgl-2.2.0.zip/download"
    3.234 D Discovered LWGL version "2.1.0" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.1.0/lwjgl-2.1.0.zip/download"
    3.234 D Discovered LWGL version "2.0.1" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.0.1/lwjgl-2.0.1.zip/download"
    3.234 D Discovered LWGL version "2.0" at "https://sourceforge.net/projects/java-game-lib/files/Official%20Releases/LWJGL%202.0/lwjgl-2.0.zip/download"
    3.234 D Loaded LWJGL list.
   30.272 D [Analytics] "Query string = v=1&tid=UA-87731965-2&cid=01ef08d5-2bda-4fe4-bfc0-007aab6ec463&t=screenview&ul=en-us&vp=694x563&sr=1920x1080&aip=1&cd=Main Window&an=MultiMC5&av=0.5.0-develop-886&cd1=&cd2=&cd3=64&cd4=64&cd5=Linux&cd6=4.4.0-57-generic&cm1=512&cm2=1024&cm3=7864&qt=30159"
   30.819 D [Analytics] "Message sent"
   60.272 D "Status JSON"  started.
   60.272 D Downloading  "http://status.mojang.com/check"
   60.353 D Download succeeded: "http://status.mojang.com/check"```
It just all of a sudden finished. I'm kinda stunned.

**Comment by peterix** [comment]
That's the current log, look at the previous one to see what happened with the download.

**Comment by MggMuggins** [comment]
Shoot. I closed the session with the terminal. Is the log stored somewhere? I didn't run it again after this happened.

**Comment by peterix** [comment]
MultiMC.0.log is the latest, increasing the number goes back.

**Comment by MggMuggins** [comment]
[MultiMC-1.log.txt](https://github.com/MultiMC/MultiMC5/files/804807/MultiMC-1.log.txt)
Here is the log file. I hope that has the information pertaining to that attempt.

**Comment by MggMuggins** [comment]
Actually, I'm fairly sure It's just my really slow internet, since I just updated at school and it seems to have worked fine. Thanks for the help anyway.

**Comment by peterix** [comment]
Ok. The download was 'running' for two hours, then failed with error 2 (`RemoteHostClosedError`) and then the second attempt finished in two minutes.

So something was holding that connection open and unusable for two hours. That's really, really weird.

**Comment by MggMuggins** [comment]
Huh. My max is 80kb/sec (Yes, this is exactly what I mean, it is actually this slow), so It's entirely conceivable that something like that happened. In any case, the issue was on my side. I'll just use library internet if I need to get a fast connection for something.
You have any idea what that might have been?

**Comment by peterix** [comment]
Well... let's see.

The TCP keepalive period  for a connection on most OSes is exactly two hours by default.
```
$ sysctl net.ipv4.tcp_keepalive_time
net.ipv4.tcp_keepalive_time = 7200
```
The connection stayed open for two hours - until the server terminated the connection (`RemoteHostClosedError`).

It looks like the server never got a request for that file.

So, I will ask again that you get Wireshark or `tcpdump` and then actually capture what's going on while it fails to update like that:
```
tcpdump -i any -s 65535 -w capture_all_the_things.pcap
```

Just make sure you aren't accessing anything you want to keep private while doing that.

**Comment by peterix** [comment]
You don't have to capture the whole thing. There's two hours of pretty much nothing interesting going on after it fails to send the request. The interesting part is the beginning.

**Comment by peterix** [comment]
And don't just post it publicly - it has your IP adress and general network traffic in it. The best thing would be to send it to me directly (peterix@gmail.com) or share it in some other direct way if the file is too big.

**Comment by MggMuggins** [comment]
Sorry, I've been putting replying to this way too long. If it's not a huge problem, I'd rather not mess with it. At this point, I don't play MC enough to make a huge time commitment worth it. If you really want to see what is happening, I can do sometime, but the next two days I probably won't be around.

**Comment by spannerman79** [comment]
@MggMuggins as someone that use to have a very flaky internet connection I can tell you now this is not an issue with MuiltiMC.  This is an issue that only you can resolve by adjusting your MTU and your RWIN/BDP values - which is outside the scope of this issue tracker.

**Comment by MggMuggins** [comment]
Thanks for the info spanner.

**Comment by MggMuggins** [comment]
Hey folks, I recently moved over to Manjaro for a number of reasons, and I simply copied my MultiMC data over again, like I did with that install. MultiMC Runs fine, but My desktop file is, once again, broken. Here is the contents of that (located at `~/.local/share/applications`:
```
[Desktop Entry]
Categories=Game;
Exec=/home/<usrid>/.local/share/multimc/MultiMC
Icon=/home/<userid>/.local/share/multimc/multimc.svg
Keywords=game, multimc, minecraft;
Name=MultiMC 5
StartupNotify=true
Terminal=false
Type=Application
X-KDE-SubstituteUID=false
```
I know the Exec works, because if I bring the app up in my application menu and run it, it runs fine. My image is broken, however. I've verified that the path to the image is correct (and I indeed have a `multimc.svg` file in the multimc folder). However, Plasma refuses to find the image...

I can ask at my distro forum as well if you'd like, since this is really outside the scope of this issue, but since I had an issue here already I thought I might as well post here.

On a side note, running MultiMC adds an icon on my dock that is different than the one that I have assigned to the file in my desktop entry. Does anyone know why this happens? Is it just a side effect of giving Qt a file and telling it that that file is the application icon? Does anyone know where in the source this is happening, or can explain it more adequately?
